src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1322,6 +1322,19 @@ static ErrorCode getRefMismatchErrorCode(TypeKind type)
                 };
         }
 
+        private static void CheckForDefaultParameterMismatch(SyntaxNode methodGroupSyntax, MethodSymbol sourceMethod, MethodSymbol targetDelegateInvoke, bool isExtensionMethod, BindingDiagnosticBag diagnostics)
+        {
+            Debug.Assert(sourceMethod.ParameterCount == targetDelegateInvoke.ParameterCount + (isExtensionMethod ? 1 : 0));
```

Comment:
Consider moving this after the calculation of `start` and asserting `sourceMethod.ParameterCount == targetDelegateInvoke.ParameterCount + start`